lib/board.rb

```diff
@@ -0,0 +1,61 @@
+require 'pry'
+class Board
+
+
+	attr_accessor :cells, :edges, :corners, :center
+
+	def initialize
+		@cells = Array.new(9," ")
+		@edges = [2,4,6,8]
+		@corners= [1,3,7,9]
+		@center= [5]
+
+	end
+
+	def reset!
+		@cells.clear
+		@cells = Array.new(9," ")
+	end
+
+	def display
+		puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+	    puts "-----------"
+	    puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+	    puts "-----------"
+	    puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+	end
+
+	def position(position)
+		user_position = @cells[position.to_i-1]
+	end
+
+	def full?
+		@cells.all? {|cell| cell == "X" || cell == "O" }
+	end
+
+	def turn_count
+		turns = @cells.select {|cell| cell == "X" || cell == "O"}
+		turns.size
+	end
+
+	def taken?(position)
+		position(position) == "X" || position(position) == "O"
+	
+	end
+
+	def valid_move?(move)
+		move.to_i > 0 && move.to_i < 10 && !taken?(move.to_i)
+	end
```

Comment:
You can simplify checking the valid integer range, and make it more semantic, by using [`Comparable#between?`](http://ruby-doc.org/core-2.2.3/Comparable.html#method-i-between-3F).
`move.to_i.between?(1,9)`